### PR TITLE
fix(shared-data): Allow mag block in the app

### PR DESCRIPTION
**Pull request by TamarZanzouri**
# Overview

closes https://opentrons.atlassian.net/browse/RQA-783.
fixes bug when uploading a protocol via the app with a mag block.

# Test Plan

- Tested via the app, uploaded `Illumina DNA Prep 24x No pause Offset Fix.py` specified in the ticket to an OT3 and it works as expected. 

# Changelog

- Added mag block logic to `getModuleDef2 in modules.ts
- Added mag block to `MODULE_TYPES` and `MODULE_MODELS` in constants.ts
- Added mag block to `ModuleType` and `ModuleModel` in types.ts
- Added mag block to all UI pages that use modules.
- Added a mag block image.

# Review requests

Did I miss anything?

# Risk assessment

low. bug fix.

### CODE REVIEW

File: shared-data/command/schemas/.json
Comment (b-cooper) [inline shared-data/command/schemas/.json:1]: whoops. this looks like an accidental addition

**Review by b-cooper — APPROVED** [review]
The Changes look good, as long as this runs fine in the app and PD, it looks good to me

**Comment by jerader** [comment]
You're gonna have to filter out the `magneticBlockV1` type from protocol designer because otherwise, it is selectable.

Screen Shot 2023-05-11 at 3 13 33 PM

**Review by jerader — APPROVED** [review]
lgtm!

**Pull request merged**